### PR TITLE
DEV-2010 send_message implementation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: krux_sqs/sqs.py [hunk 1]
```diff
@@ -84,6 +84,11 @@ def add_sqs_cli_arguments(parser, include_boto_arguments=True):
     group = get_group(parser, NAME)
 
 
+def sqs_send_message(queue_name, message):
+    """Utility function for sending SQS messages."""
+    Sqs(Boto3()).send_message(queue_name, message)
```
Comment (hanpeter) [inline krux_sqs/sqs.py]: There are number of problems with this approach.
1. This provides no way to override the authentication method for Boto. `krux-boto` object will always use default authentication method (env var).
2. This nullifies the built in caching of SQS queue objects in `krux-sqs`. With `Sqs` object, the queue will be cached and reused. This approach will re-instantiate everything.

File: krux_sqs/sqs.py [hunk 2]
```diff
@@ -192,3 +197,17 @@ def delete_messages(self, queue_name, messages):
...
             self._logger.debug('Messages list is empty. Not deleting any messages.')
+
+    def send_message(self, queue_name, message):
+        """
+        Send the given message to the given queue.
+
+        :param queue_name: :py:class:`str` Name of the queue to send messages.
+        :param message: :py:class:`dict` Message dict to send.
+        """
+        # GOTCHA: queue.delete_messages() does not handle an empty list
```
Comment (hanpeter) [inline krux_sqs/sqs.py]: Copy & paste artifact.

Also, does `send_message()` not handle the empty message body just like `delete_message()`? In other words, is this GOTCHA needed?

File: krux_sqs/sqs.py [hunk 3]
```diff
@@ -192,3 +197,17 @@ def delete_messages(self, queue_name, messages):
...
+        if message:
+            self._logger.debug('Send following message: %s', message)
+            self._get_queue(queue_name).send_message(MessageBody=simplejson.dumps(message))
```
Comment (hanpeter) [inline krux_sqs/sqs.py]: Let's provide a way to send raw message as is. What if I just wanted to send a raw string to SQS?

File: test/sqs_test.py [hunk 4]
```diff
@@ -59,3 +59,7 @@ def test_delete_messages(self):
         # let's leave it at this minimal state.
         messages = self._sqs.get_messages(self.TEST_QUEUE_NAME)
         self._sqs.delete_messages(self.TEST_QUEUE_NAME, messages)
+
+    def test_send_message(self):
```
Comment (hanpeter) [inline test/sqs_test.py:63]: Docstring, please.

Also, if you are not implementing a full unit test for `send_message()`, please leave a TODO to fix later.

**Comment by MikalaiT** [comment]
Hi @hanpeter ,

I see you are the author of this library. So, could you fix this changes using the right approach or implement the method for message sending?

CC:@jib

File: test/sqs_test.py (hunk 4 above)
Comment (ghost) [inline test/sqs_test.py:63]: done

File: krux_sqs/sqs.py (hunk 1 above)
Comment (ghost) [inline krux_sqs/sqs.py]: @hanpeter could you please provide the correct way to work with SQS? Thanks

**Comment by hanpeter** [comment]
@alekseyC64 & @MikalaiT: I have made the necessary changes to match the coding style here. Please take a look.

In terms of how to send messages, I would expect something like this:

``` python
class Application(object):

    def __init__(self):
        boto = Boto(
            access_key='fake-access-key',
            secret_key='fake-secret-key',
            region='us-east-1',
        )
        self._sqs = Sqs(
            boto=boto()
        )

    def run(self):
        # Do some magic

        if send_message:
            self._sqs.send_messages(queue_name='my-test-queue', messages=['msg1', 'msg2'])

        # Do more magic
```

This way, you will have a way to adjust authorization in your application as well as the AWS data center. Also, this will allow you to utilize the caching of this library, so you won't need to fetch the queue object every time. Let me know if this doesn't make sense.

**Comment by ghost** [comment]
@hanpeter LGTM, could you please review usage of this in https://github.com/krux/rb_dataconsole/pull/2088/files ? thanks

**Comment by ghost** [comment]
:+1:

**Pull request merged**